doc/ExpertOptions.md

```diff
@@ -1555,3 +1556,66 @@ dx describe /builds/1.02/applets/hello --json --details | jq '.details | .wdlSou
+```
+
+Example: adding authorized users to a global workflow
+```
+dx add users globalworkflow-<workflow name> <user-xxxx | org-yyyy | PUBLIC>
+```
+
+See [Limitations](#limitations) below for more details on which parts of the workflow will be automatically included in the global workflow.
+
+## Recommendations
+
+Developers should not store any credentials (passwords, keys, etc.) in the source code of the global workflow, as authorized users will have permission to download and view any applets referenced in the global workflow.
+
+Developers should use simple types for input/output interfaces for global workflows for better platform experience.
+
+For better portability across projects where the workflow will be run, developers should specify runtime resources using memory, disk, CPU requirements (not hard-coded instance types using the key `dx_instance_type`). Developers should compile WDL workflows with `-instanceTypeSelection dynamic`. Expect slightly longer runtimes due to the need to pick instance types at runtime, but this ensures the workflow will always use instance types that are supported for the project where it runs.
+
+For informational purposes, developers should include a reference to a git repo commit containing the original source code. They can add this to the "description" metadata field of the global workflow.
+
+## Limitations
+
+Publishing a dxCompiler workflow as a global workflow is currently only supported for WDL.
+
+The global workflow will currently only support a single region (matching the region in which the original workflow was compiled).
+
+Some dependencies of the original workflow will be automatically included in the global workflow, i.e. they will be cloned into the global workflow's resource container and authorized users of the global workflow will not require additional permissions. These include
+- Applets and sub-workflows that were part of the original workflow
+- Native applets referenced by the workflow
+- Docker images that are stored as platform files
+
+Some dependencies of the original workflow will not be automatically included in the global workflow, i.e. the user may need additional permissions to access them. These include
+- Native apps referenced by the workflow (user needs permission to use the apps)
+- Files referenced by file paths in workflow inputs or body (user needs access to the files)
+- Credentials file for a private Docker registry (user needs access to the file)
+- Docker images in external registries, or dynamically specified at runtime (these will be pulled at runtime)
+- Hard-coded `dx_instance_type` (runtime project needs to support the instance type; using numeric resource requirements is preferred)
+
+<!-- TODO mention URL when the UI supports global workflows -->
+Any usage of the above in a workflow (including in its tasks and sub-workflows) will produce a warning in the workflow's `description` metadata field, which can be viewed using:
```

Comment:
Just to confirm, the warning will also be produced for docker images from external registries (public images)?

Reply:
Yes, this is already covered in this line
"- Docker images in external registries, or dynamically specified at runtime (these will be pulled at runtime)"